src/hb-aat-map.cc

```diff
@@ -81,7 +90,7 @@ hb_aat_map_builder_t::compile (hb_aat_map_t  &m)
     features.qsort ();
     unsigned int j = 0;
     for (unsigned int i = 1; i < features.length; i++)
-      if (features[i].type != features[j].type)
+      if (features[i].type != features[j].type || (!features[i].is_exclusive && (features[i].setting ^ features[j].setting) > 1))
```

Comment:
I don't understand the check for xor result `> 1`.  Can you clarify please?

Reply:
Per https://developer.apple.com/fonts/TrueType-Reference-Manual/RM09/AppendixF.html:

> For nonexclusive features, selectors come in pairs consisting of an even number (0, 2, 4,…) and the odd number following it (1, 3, 5,…). The even number indicates that the selector is on, and the odd number that it is explicitly off.

Or to look at it another way, the low bit of the selector indicates whether to turn this setting of the given type on (low bit = 0) or off (low bit = 1).

So if we have multiple feature requests for the same type, and they differ *only* in the low bit, they're referring to the same setting and we only want to keep the first one; but if they differ in any higher bits (i.e. xor > 1) then they're distinct settings (like Common vs Discretionary, both under the Ligatures type) and we want to keep both.

Reply:
FWIW, I rewrote this in the `((a & ~1) != (b & ~1))` form, as that seems a bit less cryptic.